src/internals/types.ts

```diff
@@ -95,6 +95,8 @@ export type PromiseOrPlain<T, Decider extends AnyFn> =
 
 export type NoPromise<T> = T extends Promise<unknown> ? never : T;
 
+export type UnwrapPromise<T> = T extends Promise<infer U> ? U : T;
```

Comment:
Can you use `Awaited<T>`? https://www.typescriptlang.org/docs/handbook/utility-types.html

It's not 100% the same thing (it also unwraps nested promises recursively) but it might be what you want to use nonetheless

Reply:
Nice, thanks!

Reply:
You forgot to remove the utility class 😁